tools/certs/generate-istioctl-certs.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Generates files for istioctl's --cert-dir directory.
+
+set -o errexit
+
+if [[ "x$DEBUG_SHELL_SCRIPT" != "x" ]]; then
+   set -x
+fi
+
+if [ -z "${ISTIO_NAMESPACE}" ]; then
+   export ISTIO_NAMESPACE=istio-system
+   echo '$ISTIO_NAMESPACE not set, defaulting to' $ISTIO_NAMESPACE
+fi
+
+if [ -z "${CERT_DIR}" ]; then
+   export CERT_DIR=istioctl-certs
+   echo '$CERT_DIR not set, defaulting to' $CERT_DIR
+fi
+
+echo ""
+
+mkdir -p ${CERT_DIR}
+kubectl get secret istio-ca-secret -n ${ISTIO_NAMESPACE} -o "jsonpath={.data['ca-cert\.pem']}" | base64 -d > ${CERT_DIR}/k8s-root-cert.pem
```

Comment:
this will be a problem I think for central istiod. I believe you want to retrieve secretes for `cacerts`. but worth a double check.